.github/workflows/My-CI-Pipeline.yml

```
@@ -16,4 +16,32 @@ jobs:
     - uses: actions/checkout@v2
     - run: docker build --tag dani --target test .
     - run: docker run --env-file .env.test dani
-     
\ No newline at end of file
+
+ build-two:
+    name: build production image and push to dockerhub  
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - if: ${{ github.event_name == 'push' && github.event.action == 'm8' }}
+        run: echo "publishing!"
+      - uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}    
+      - run: docker build --target production --tag daniphilip/hello-world:prod .
+      - run: docker push daniphilip/hello-world:prod
+      
+ build-three:
+    name: deploying and running onto heroku
+    runs-on: ubuntu-latest
```

Note that this will want to rely on the `build` job just like `build-two` does, and that it also only wants to run on main once we've tested it; we can do that with:
```suggestion
    runs-on: ubuntu-latest
    needs: build-two
    if: github.ref == 'refs/heads/main' && github.event_name != 'pull_request' # include this line if you want to limit the deployment to the main branch
```